### ISSUE TITLE
Add support for additional xgboost objectives

### ISSUE THREAD

**Issue by EmilHvitfeldt** [tidymodels/tidypredict#184]
Add support for the following objectives in `build_fit_formula_xgb()`:

**Regression:**
- `reg:squaredlogerror` (needs positive labels)
- `reg:gamma` (needs positive labels)
- `reg:pseudohubererror`
- `reg:absoluteerror`

**Classification:**
- `binary:hinge`

**Multiclass** (requires special handling for multiple outputs per row):
- `multi:softmax` (returns class labels)
- `multi:softprob` (returns probabilities per class)

Also update the error message at `R/model-xgboost.R:243` which is outdated - it's missing `count:poisson` and `reg:tweedie` from the supported list.
